vim/vimrc

```diff
@@ -5,49 +5,51 @@ set nocompatible
 
 
 """ Vim Plugins and Management """
-" Plug setup.
-call plug#begin('~/.vim/plugged')
-	" Formula.
-	"  Github Repo.
-	"  Plug 'name/project'
-	Plug 'vim-scripts/VimOutliner'", { 'for': 'otl' }
-	Plug 'Aviator45003/vim-script-refactor'
-	Plug 'christoomey/vim-tmux-navigator'
-	Plug 'dahu/vim-fanfingtastic'
-	Plug 'saltstack/salt-vim'
-	Plug 'majutsushi/tagbar'
-	Plug 'tpope/vim-eunuch'
-"	Plug 'kien/ctrlp.vim'
-	Plug 'szw/vim-ctrlspace'
-	Plug 'joonty/vdebug', { 'for': 'php' }
-	Plug 'elzr/vim-json'
-	Plug 'dahu/vim-lotr'
-call plug#end()
-" Plug end.
+" Ensure that vim-plug exists
+if !empty(glob('~/.vim/autoload/plug.vim'))
+	" Plug setup.
+	call plug#begin('~/.vim/plugged')
+		" Formula.
+		"  Github Repo.
+		"  Plug 'name/project'
+		Plug 'Aviator45003/vim-script-refactor'
+		Plug 'christoomey/vim-tmux-navigator'
+		Plug 'dahu/vim-fanfingtastic'
+		Plug 'dahu/vim-lotr'
+		Plug 'elzr/vim-json'
+		Plug 'joonty/vdebug', { 'for': 'php' }
+"		Plug 'kien/ctrlp.vim'
+		Plug 'majutsushi/tagbar'
+		Plug 'saltstack/salt-vim'
+		Plug 'szw/vim-ctrlspace'
+		Plug 'tpope/vim-eunuch'
+		Plug 'vim-scripts/VimOutliner'", { 'for': 'otl' }
+	call plug#end()
+endif
 
 
 " For backup versioning.
-if has("vms")
+if has('vms')
 	set nobackup    " do not keep a backup file, use versions instead
 else
 	set backup      " keep a backup file
 endif
 
 
 """ Special File Setup """
+" Because we also want our backup files to not be all over.
+set backupdir=~/.vim/backups//
 " Because it gets annoying when you are editing a file in Dropbox and the
 " .swp file gets shared...
-set dir=~/.vim/swaps
-" Because we also want our backup files to not be all over.
-set backupdir=~/.vim/backups
+set directory=~/.vim/swaps//
 " Undo files.
-set undodir=~/.vim/undo
+set undodir=~/.vim/undo//
```

Comment:
"Use absolute file paths when creating temporary files".  It prevents swap file collisions from occuring when two files with the same filenames but different paths are open simultaneously. 

It changes the names of the swap files to the absolute path of the file being edited.  Slashes in the path are replaced with moduli.